viewer/packages/camera-manager/src/Flexible/FlexibleCameraManager.ts

```diff
@@ -0,0 +1,551 @@
+  private _nearAndFarNeedsUpdate = false;
+  private _isEnabled = true;
+
+  // For the wheel event
+  private _prevTime = 0;
+  private readonly _prevCoords = new Vector2();
+
+  //================================================
+  // INSTANCE FIELDS: Events
+  //================================================
+
+  private readonly _raycastCallback: RaycastCallback;
+
+  //================================================
+  // CONSTRUCTOR
+  //================================================
+
+  constructor(
+    domElement: HTMLElement,
+    inputHandler: InputHandler,
+    raycastCallback: RaycastCallback,
+    camera?: PerspectiveCamera,
+    scene?: Scene
+  ) {
+    this._camera = camera ?? new PerspectiveCamera(45, undefined, 0.1, 10000);
+    this._controls = new FlexibleControls(this.camera, domElement, new FlexibleControlsOptions());
+    this._domElement = domElement;
+    this._inputHandler = inputHandler;
+    this._raycastCallback = raycastCallback;
+    if (scene) {
+      this._markers = new FlexibleCameraMarkers(scene);
+    }
+
+    this.addEventListeners();
+    this._controls.addEventListeners(); // After this.addEventListeners();
+
+    this.controls.addEventListener('cameraChange', event => {
+      const { position, target } = event.camera;
+      this._events.cameraChange.fire(position.clone(), target.clone());
+      this._nearAndFarNeedsUpdate = true;
+      if (this._markers) {
+        this._markers.update(this);
+      }
+    });
+
+    this.isEnabled = true;
+    this._stopEventTrigger = new DebouncedCameraStopEventTrigger(this);
+  }
+
+  //================================================
+  // IMPLEMENTATION OF CameraManager (In correct order)
+  //================================================
+
+  public getCamera(): PerspectiveCamera {
+    return this.camera;
+  }
+
+  public getCameraState(): Required<CameraState> {
+    return {
+      position: this.getPosition(),
+      rotation: this.camera.quaternion.clone(),
+      target: this.getTarget()
+    };
+  }
+
+  /**
+   * Sets camera state. All parameters are optional. Rotation and target can't be set at the same time,
+   * if so, error will be thrown. Set rotation is preserved until next call of setCameraState with
+   * empty rotation field.
+   * @param state Camera state.
+   * **/
+  public setCameraState(state: CameraState): void {
+    if (state.rotation && state.target) {
+      throw new Error(`Rotation and target can't be set at the same time`);
+    }
+    const newPosition = state.position ?? this.getPosition();
+    const newRotation = (state.target ? new Quaternion() : state.rotation) ?? new Quaternion();
+    const newTarget =
+      state.target ??
+      (state.rotation
+        ? CameraManagerHelper.calculateNewTargetFromRotation(this.camera, state.rotation, this.getTarget(), newPosition)
+        : this.getTarget());
+
+    if (this.controls.isEnabled) {
+      this.controls.cameraRawRotation.copy(newRotation);
+    }
```

Comment:
I don't understand the purpose of these lines - `newRotation` will be just the identity rotation if no rotation was supplied?